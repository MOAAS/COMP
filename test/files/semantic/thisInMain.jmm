import io.println(int) void;
import Quicksort;

class Simple extends Quicksort{
    int simpleVar;
    int verySimpleVar;


    public static void main(String[] arg){
        int a;
        int b;
        boolean c;
        int[] d;
        d = new int[3];
        c = false;
        a = 2 + this.getInt();
        b= 3;
        if (!(this.getInt() < 3) && this.getBool())
         a = 1;
        else b = 1;
    }

    public int getInt(){
      return 3;
    }
    public boolean getBool(){
      return true;
    }
 }