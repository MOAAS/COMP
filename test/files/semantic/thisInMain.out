Error: Cannot use "this" in static function at line 16, column 17.
Error: Cannot use "this" in static function at line 18, column 15.
Error: Cannot use "this" in static function at line 18, column 37.
